cms/djangoapps/contentstore/management/commands/import_content_library.py

```diff
@@ -0,0 +1,133 @@
+"""
+Script for importing a content library from a tar.gz file
+"""
+from __future__ import print_function
+
+import base64
+import os
+import tarfile
+
+from django.conf import settings
+from django.contrib.auth.models import User
+from django.core.exceptions import SuspiciousOperation
+from django.core.management.base import BaseCommand, CommandError
+from lxml import etree
+from opaque_keys.edx.keys import CourseKey
+from path import Path
+from xmodule.contentstore.django import contentstore
+from xmodule.modulestore import ModuleStoreEnum
+from xmodule.modulestore.django import modulestore
+from xmodule.modulestore.exceptions import DuplicateCourseError
+from xmodule.modulestore.xml_importer import import_library_from_xml
+
+from cms.djangoapps.contentstore.utils import add_instructor
+from openedx.core.lib.extract_tar import safetar_extractall
+
+
+class Command(BaseCommand):
+    """
+    Import the specified content library archive.
+    """
+    help = 'Import the specified content library into mongo'
+
+    def add_arguments(self, parser):
+        parser.add_argument('archive_path')
+        parser.add_argument('owner_username')
+
+    def handle(self, *args, **options):
+        """
+        Given a content library archive path, import the corresponding course to mongo.
+        """
+
+        archive_path = options['archive_path']
+        username = options['owner_username']
+
+        data_root = Path(settings.GITHUB_REPO_ROOT)
+        subdir = base64.urlsafe_b64encode(repr(archive_path.split('/')[-1]))
+        course_dir = data_root / subdir
+
+        # Extract library archive
+        tar_file = tarfile.open(archive_path)
+        try:
+            safetar_extractall(tar_file, (course_dir + u'/').encode(u'utf-8'))
+        except SuspiciousOperation as exc:
+            raise CommandError(u'\n=== Course import %s: Unsafe tar file - %s\n', archive_path, exc.args[0])
+        finally:
+            tar_file.close()
+
+        # Find the required XML file
+        abs_xml_path = _get_dir_for_filename(course_dir, 'library.xml')
```

Comment:
Are there cases where `library.xml` won't be in the root of the archive?

Reply:
Apparently there are, this is another instance of [code inherited from Studio](https://github.com/edx/edx-platform/blob/master/cms/djangoapps/contentstore/tasks.py#L855).

I haven't come across this in my testing, maybe this deals with an old format? Or maybe this is just necessary for importing courses, as the code deals with both and `course.xml` is mentioned explicitly.

I'll remove this in a separate commit so it can be reverted if necessary.